std/array.d

```diff
@@ -778,40 +778,62 @@ private auto arrayAllocImpl(bool minimallyInitialized, T, I...)(I sizes) nothrow
 }
 
 // overlap
-/*
-NOTE: Undocumented for now, overlap does not yet work with ctfe.
+
+
+/++
 Returns the overlapping portion, if any, of two arrays. Unlike $(D
-equal), $(D overlap) only compares the pointers in the ranges, not the
-values referred by them. If $(D r1) and $(D r2) have an overlapping
-slice, returns that slice. Otherwise, returns the null slice.
-*/
-auto overlap(T, U)(T[] r1, U[] r2) @trusted pure nothrow
-if (is(typeof(r1.ptr < r2.ptr) == bool))
+equal), $(D overlap) only compares the pointers and lengths in the
+ranges, not the values referred by them. If $(D r1) and $(D r2) have an
+overlapping slice, returns that slice. Otherwise, returns the null
+slice.
++/
+
+CommonType!(T[], U[]) overlap(T, U)(T[] a, U[] b) @trusted
+if (is(typeof(a.ptr < b.ptr) == bool))
 {
-    import std.algorithm.comparison : min, max;
-    auto b = max(r1.ptr, r2.ptr);
-    auto e = min(r1.ptr + r1.length, r2.ptr + r2.length);
-    return b < e ? b[0 .. e - b] : null;
+    import std.algorithm.comparison : min;
+    
+    auto end = min(a.ptr + a.length, b.ptr + b.length);
+    //CTFE requires pairing pointer comparisons, which forces a
```

Comment:
Space after `//` :). Not sure I get what "pairing pointer comparisons" means... At any rate, if this is worth it (probably not) we can always use `if (__ctfe)`.